### ISSUE TITLE
multi_gpu_model fails with custom merge layers

### ISSUE THREAD

**Issue by dulaku** [keras-team/keras#12764]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Debian 9 Stretch
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.13.1
- Keras version:  2.2.4
- Python version:  3.5
- CUDA/cuDNN version:  10.0/7
- GPU model and memory:  NVIDIA RTX 2080 7951MiB x2

**Describe the current behavior**  
When using `keras.utils.multi_gpu_model()` to parallelize a model that includes merge layers that involve addition, in a model using a custom loss function that calls `keras.backend.gradients()`, model compilation fails because `keras.backend.gradients()` returns `[None]` for the addition operation. The example script below exits with code 1.

This isn't strictly limited to layers that extend `keras.layers.merge._Merge` - I ran into the same issue when using a combination of `keras.layers.multiply` and `keras.layers.add` layers to implement the same functionality, but this seems a bit more readable. I've pointed out as much of the details in the script below, but I can't follow the issue past one of the Tensorflow functions being used. Let me know if there's any more information I can supply!

**Describe the expected behavior**  
Model compilation succeeds. The example script below exits with code 0. Ideally, the model would also be trained on multiple GPUs, but this isn't part of the script.

**Code to reproduce the issue**  
A sample script is available at this [gist](https://gist.github.com/dulaku/ce2b82c05d61273f67c379a193b4051f).

**Other info / logs**  
A traceback is available at this [gist](https://gist.github.com/dulaku/f304c6261e6ea2e50ee09bd9c0d05fdf). Additionally, the Tensorflow op that generates the issue in the sample script is:

```name: "random_weighted_average_1/add"
op: "Add"
input: "random_weighted_average_1/mul"
input: "random_weighted_average_1/mul_1"
attr {
  key: "T"
  value {
    type: DT_FLOAT
  }
}
